File: 11_Container_With_Most_Water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int i = 0;
        int j = height.size()-1;
        int maxcap = INT_MIN;
        
        while(i <j){
            int temp = (j-i) * min(height[i],height[j]);
            maxcap = max(temp,maxcap);
            if(height[i] < height[j]){
                i++;
            }else{
                j--;
            }
        }
        return maxcap;
    }
};

File: 11_Container_With_Most_Water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l, r = 0, len(height)-1
        water = 0
        maxWater = 0
        while l < r:
            water = (r-l) * min(height[l],height[r])
            maxWater = max(maxWater,water)
            if height[l] < height[r]:
                l += 1
            else:
                r -= 1
        return maxWater